### PR TITLE
Fix siwe handling

### PR DESCRIPTION
**Pull request by kristoferlund**
This PR should ensure stable sign in functionality. Now taking into account 401 errors from backend. When a 401 error is received, any auth information in frontend is deleted to trigger new login.

1. Ensure auth modal appears after rejecting eth signature
2. User not being logged in does not mean auth modal should be shown, if wallet is not connected for example
3. Force new login when receiving 401 response

### CODE REVIEW

File: src/utils/auth.ts [hunk 1]
```diff
@@ -2,6 +2,22 @@ import { SiweMessage } from 'siwe'
 import { User } from '@/types/user/User'
 import { axiosInstance } from './axiosInstance'
 
+axiosInstance.interceptors.response.use(
+  function (response) {
+    return response
+  },
+  function (error) {
+    if (error.response && error.response.status === 401) {
+      window.localStorage.removeItem('auth')
```
Comment (mmahdigh) [inline src/utils/auth.ts:11]: I think you can reuse the `logout` function here. They seem to be doing the same thing.

File: src/utils/auth.ts [hunk 2]
```diff
@@ -50,10 +66,9 @@ export const login = async (
     window.localStorage.setItem('auth', verifyRes.data)
     window.localStorage.setItem('loggedInAddress', address)
     axiosInstance.defaults.headers.common['auth'] = verifyRes.data
-    window.location.reload()
     return verifyRes
   } catch (error) {
-    console.error('ERROR', error)
+    throw error
```
Comment (mmahdigh) [inline src/utils/auth.ts:71]: Not a big deal, but if you just want to throw the caught error, you can just get rid of the try-catch block.

File: src/utils/auth.ts [hunk 3]
```diff
@@ -23,7 +39,7 @@ const fetchNonce = async () => {
 export const login = async (
   chainId: number,
   address: string,
-  signFunction: (args?: any) => Promise<`0x${string}`>
+  signFunction: (args?: any) => Promise<`0x${string}` | undefined>
```
Comment (mmahdigh) [inline src/utils/auth.ts:42]: What's the scenario that the sign function would return `undefined`? And if it does, we probably need to handle that scenario in this `login` function.

Comment (kristoferlund) [reply src/utils/auth.ts:42]: Good catch. There is no scenario. I was planning other solution first but realised it is better to just let sigFunction throw if, for instance, user rejects signature. Removing.

File: src/utils/auth.ts (hunk 2 above)
Comment (kristoferlund) [reply src/utils/auth.ts]: Fixed

**Pull request merged**